### ISSUE TITLE
Add tabbed sessions and keyboard shortcuts to navigate between them

### ISSUE THREAD

**Issue by mshostack** [continuedev/continue#1858]
### Validations

- [X] I believe this is a way to improve. I'll try to join the [Continue Discord](https://discord.gg/NWtdYexhMs) for questions
- [ ] I'm not able to find an [open issue](https://github.com/continuedev/continue/issues?q=is%3Aopen+is%3Aissue+label%3Aenhancement) that requests the same enhancement

### Problem

When using Continue to generate a list of tasks or high-level outline of something I then want to dive deeper on, I want the ability to create a new session tab(s) to preserve the context and accessibility of the previous response without having to use "Last Session" which is limiting and clunky.

### Solution

1. These tabs should be created using a button or keyboard shortcut
2. The initial tab I'm on that I want to preserve should add a little onHover button similar to what appears when you mouseover a codeblock, but instead of options to run in terminal, insert, or copy, it should let me either:
  a. Start a new tab session using the selected text (which may include just text or other formatted items like codeblocks)
  b. Enter a new starting prompt for the new tab, and then insert the selected text/context below it. Think "Create a detailed step by step approach for implementing the below step from a refactoring plan"

**Comment by ferenci84** [comment]
@mshostack I wanted to find if there is an issue for adding tabs feature (before I open a new issue) like this:

<img width="409" alt="Image" src="https://github.com/user-attachments/assets/89e13788-5a2b-42cb-ad32-6c29eeadf586" />

Do your suggestion include this enhancing with an additional feature? Can you show a use-case example?

What I also find useful in other similar applications is the "fork" feature:

<img width="1002" alt="Image" src="https://github.com/user-attachments/assets/82e45286-223e-4c3f-a79b-aedd62d9ed3f" />

Combined with tabs, it would be super-helpful for example when you have a side question related to the current tab, so it helps to have a context, but you don't want to bloat the current session.

Is your suggestion similar to this?

**Comment by mshostack** [comment]
Yes! Both of these examples you gave capture the essence of this.

As an example use case, lets say I generate an outline of creating a UI. I may need to iterate on each step, or may want to break out more context for each step. But to keep using the same context window for all steps, with any iteration on them and back and forth I do, would simply be a waste of tokens. So popping a given step from a numbered list out to a new tab is what I'd want to do.

Importantly though, selected text may not be enough on its own for a proper prompt. So I'd need the ability to say, take selected text, pre-populate it into a "New Tab Prompt" and let me continue modifying that prompt before creating the new tab with it.
